src/alley/wp/alleyvate/features/class-disable-comments.php

```diff
@@ -54,6 +55,31 @@ public static function action__admin_bar_menu( \WP_Admin_Bar $wp_admin_bar ): vo
 		$wp_admin_bar->remove_node( 'comments' );
 	}
 
+	/**
+	 * Removes blocks related to core/comments from the admin block selector.
+	 * 
+	 * JavaScript is used to selectively remove blocks from the editor.
+	 * The PHP filter for allowed blocks passes ‘true’ to allow all blocks by default,
+	 * so you can’t get the full list of blocks and selectively remove them.
+	 * https://developer.wordpress.org/news/2024/01/how-to-disable-specific-blocks-in-wordpress/#disable-blocks-with-php
+	 */
+	public static function action__admin_footer(): void {
+		echo "
+			<script>
+			if (typeof wp?.domReady === 'function') {
+				wp.domReady( () => {
+					if (typeof wp?.blocks?.unregisterBlockType === 'function') {
+						// Unregister blocks related to core comments.
+						wp.blocks.unregisterBlockType( 'core/comments' );
+						wp.blocks.unregisterBlockType( 'core/post-comments-form' );
+						wp.blocks.unregisterBlockType( 'core/comments-query-loop' );
+						wp.blocks.unregisterBlockType( 'core/latest-comments' );
+					}
+				} );
+			}
```

Comment:
We've got a mix of coding styles here between what we typically do in JS and the WP standards. Let's standardize on our typical JS syntax:
```suggestion
			if (typeof wp?.domReady === 'function') {
				wp.domReady(() => {
					if (typeof wp?.blocks?.unregisterBlockType === 'function') {
						// Unregister blocks related to core comments.
						wp.blocks.unregisterBlockType('core/comments');
						wp.blocks.unregisterBlockType('core/post-comments-form');
						wp.blocks.unregisterBlockType('core/comments-query-loop');
						wp.blocks.unregisterBlockType('core/latest-comments');
					}
				});
			}
```
Unfortunately, ESLint isn't going to catch this because it's JS-in-PHP, but it would be better to have consistency even if the linter isn't complaining about it.